src/test/kotlin/com/pubnub/api/presence/PresenceTest.kt

```diff
@@ -94,20 +97,70 @@ internal class PresenceTest {
         assertThat(presence, Matchers.isA(PresenceNoOp::class.java))
     }
 
+    @Test
+    fun `Leave events not created when suppressLeaveEvents is true and heartbeat interval is 0`() {
+        // given
+        val presence = Presence.create(
+            listenerManager = listenerManager, heartbeatInterval = 0.seconds, suppressLeaveEvents = true,
+            leaveProvider = object : LeaveProvider {
+                override fun getLeaveRemoteAction(
+                    channels: Set<String>,
+                    channelGroups: Set<String>
+                ): RemoteAction<Boolean> {
+                    throw IllegalStateException("Leave events should not be created!")
+                }
+            },
+        )
+
+        // when
+        presence.joined(setOf("abc"))
+        presence.leftAll()
+
+        // then
+        // no exception
+    }
+
+    @Test
+    fun `Leave events created when suppressLeaveEvents is false and heartbeat interval is 0`() {
+        // given
+        val success = AtomicBoolean(false)
+        val presence = Presence.create(
+            listenerManager = listenerManager, heartbeatInterval = 0.seconds, suppressLeaveEvents = false,
+            leaveProvider = object : LeaveProvider {
```

Comment:
We could use mock here. What do you think?

Reply:
done